src/fs/root/tree.ts

```diff
@@ -256,44 +260,47 @@ type PathKey = { path: DistinctivePath, key: string }
 
 
 async function findBareNameFilter(
-  map: Record<string, PrivateTree>,
+  map: Record<string, PrivateNode>,
   path: DistinctivePath
 ): Promise<Maybe<BareNameFilter>> {
   const bareNameFilterId = await identifiers.bareNameFilter({ path })
   const bareNameFilter: Maybe<BareNameFilter> = await storage.getItem(bareNameFilterId)
   if (bareNameFilter) return bareNameFilter
 
-  const [treePath, tree] = findPrivateTree(map, path)
-  if (!tree) return null
+  const [nodePath, node] = findPrivateNode(map, path)
+  if (!node) return null
 
   const unwrappedPath = pathing.unwrap(path)
-  const relativePath = unwrappedPath.slice(pathing.unwrap(treePath).length)
-  if (!tree.exists(relativePath)) {
-    if (pathing.isDirectory(path)) await tree.mkdir(relativePath)
-    else await tree.add(relativePath, "")
+  const relativePath = unwrappedPath.slice(pathing.unwrap(nodePath).length)
+  if(PrivateFile.instanceOf(node)) {
+    return relativePath.length === 0 ? node.header.bareNameFilter : null
+  }
+  if (!node.exists(relativePath)) {
+    if (pathing.isDirectory(path)) await node.mkdir(relativePath)
+    else await node.add(relativePath, "")
```

Comment:
What's the reason we're adding this here? 👀

Reply:
```ts
if (!node.exists(relativePath)) {
    if (pathing.isDirectory(path)) await node.mkdir(relativePath)
    else await node.add(relativePath, "")
```
this part already existed before the PR
```ts
if(PrivateFile.instanceOf(node)) {
    return relativePath.length === 0 ? node.header.bareNameFilter : null
```
this part I added and is to prevent trying to access relative paths off of a file (which would not exist)

Reply:
Ah gotcha 👍 
My bad, I should've added comments to this 😅